harvesters/src/main/java/org/fao/geonet/kernel/harvest/harvester/csw/Harvester.java

```diff
@@ -99,6 +100,9 @@ public Harvester(AtomicBoolean cancelMonitor, Logger log, ServiceContext context
         this.cancelMonitor = cancelMonitor;
         this.log = log;
         this.context = context;
+        if(!(context instanceof ServiceContext.AppHandlerServiceContext)){
+            log.debug("Harvester expects to work with shared \"harvester\" service context, '"+context.getService()+"' provided.");
```

Comment:
@josegar74 I have been using a debug statement here in order to have a check for logs and debugger.

All the harvesters should be operating with a shared AppHandlerServiceContext as described above. But I would ask that we test each one.